src/components/ClinicalFeatureTable.js

```diff
@@ -0,0 +1,287 @@
+  let fileInput = useRef(null);
+
+  const [editableClinicalFeatureDefinitions, setEditableClinicalFeatureDefinitions] = useState({
+    // "Age": { "Type": CLINCAL_FEATURE_TYPES[0], "Encoding": "None" },
+    // "Gender": { "Type": CLINCAL_FEATURE_TYPES[0], "Encoding": "Categorical" }
+  });
+
+  const handleInputChange = (e, feature_name, feature_type) => {
+    let editableClinicalFeatureDefinitionsToUpdate = { ...editableClinicalFeatureDefinitions };
+    editableClinicalFeatureDefinitionsToUpdate[feature_name][feature_type] = e.target.value;
+    setEditableClinicalFeatureDefinitions(editableClinicalFeatureDefinitionsToUpdate)
+  };
+
+  useEffect(() => {
+    // Load clinical features the first time the component is rendered
+    loadClinicalFeatures();
+  }, []); // Empty dependency array ensures the effect runs only once
+
+
+  const toggleManualLabelling = () => {
+    console.log("toggleManualLabelling");
+    setisManualClinFeaturesOpen((open) => !open);
+    setisAutoClinFeaturesOpen(false);
+    loadClinicalFeatures();
+  };
+
+  const toggleAutoLabelling = () => {
+    setisAutoClinFeaturesOpen((open) => !open);
+    setisManualClinFeaturesOpen(false);
+  };
+
+  const handleSaveClinicalFeaturesClick = async (e) => {
+    if (isAutoClinFeaturesOpen) {
+      toggleManualLabelling();
+      setclinicalFeatureFileMessage(null);
+      setisClinicalFeatureFileValid(null);
+      fileInput.current.value = '';
+    }
+
+    await Backend.saveClinicalFeatures(
+      keycloak.token,
+      editableClinicalFeatures,
+    );
+
+    await Backend.deleteClinicalFeatures();
+  };
+
+  const loadClinicalFeatures = async () => {
+    let clinicalFeaturesToUpdate = { ...editableClinicalFeatures };
+
+    let clinicalFeatures = await Backend.loadClinicalFeatures(keycloak.token, dataPoints)
+
+    for (let patientID in clinicalFeatures) {
+      clinicalFeaturesToUpdate[patientID] = clinicalFeatures[patientID];
+    }
+
+    if (Object.keys(clinicalFeatures).length > 0) {
+      console.log("updating clinical features", clinicalFeaturesToUpdate)
+      setEditableClinicalFeatures(clinicalFeaturesToUpdate);
+    }
+  }
+
+  const updateEditableClinicalFeatures = (labels) => {
+    let clinicalFeaturesToUpdate = { ...editableClinicalFeatures };
+
+    for (let patientID in labels) {
+      if (patientID in editableClinicalFeatures) {
+        console.log(patientID)
+        clinicalFeaturesToUpdate[patientID] = labels[patientID];
+      }
+    }
+
+    setEditableClinicalFeatures(clinicalFeaturesToUpdate);
+  };
+
+  const handleFileInputChange = async () => {
+    let [isValid, message, clinicalFeatures] = await validateLabelOrClinicalFeaturesFile(
+      fileInput.current.files[0],
+      dataPoints
+    );
+
+    if (isValid) {
+      updateEditableClinicalFeatures(clinicalFeatures);
+      console.log(clinicalFeatures);
+    }
+    setisClinicalFeatureFileValid(isValid);
+    setclinicalFeatureFileMessage(message);
+  };
+
+  useEffect(() => {
```

Comment:
Same comment as for the other useEffect hook above

Reply:
works for me - don't know the react dev conventions yet!